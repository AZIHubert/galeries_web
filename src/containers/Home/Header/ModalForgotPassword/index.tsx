import { useFormik } from 'formik';
import * as React from 'react';

import Field from '#components/Field';

import { resetPasswordSchema } from '#helpers/schemas';

import ModalContainer from '#components/ModalContainer';
import GradientButton from '#components/GradientButton';

import { resendResetPassword } from '#helpers/api';

import {
  CancelButton,
  CancelButtonContainer,
} from './styles';

interface ModalForgotPasswordI {
  setCurrentEmail: React.Dispatch<React.SetStateAction<string>>;
  setError: React.Dispatch<React.SetStateAction<string>>;
  setForgotPassword: React.Dispatch<React.SetStateAction<boolean>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  switchModal: () => void;
  loading: boolean;
}

const initialValues = {
  email: '',
};

const ModalForgotPassword = ({
  setCurrentEmail,
  setError,
  setForgotPassword,
  setLoading,
  switchModal,
  loading,
}: ModalForgotPasswordI) => {
  const formik = useFormik({
    initialValues,
    onSubmit: async (value, { setFieldError }) => {
      if (!loading) {
        setLoading(true);
        try {
          await resendResetPassword(value);
          setCurrentEmail(value.email);
          switchModal();
        } catch (err) {
          if (err.response) {
            if (err.status === 500) {
              setError('Something went wrong. Please try again');
            } else {
              const { errors } = err.response.data;
              if (typeof errors === 'object') {
                Object.keys(errors).map((error) => setFieldError(error, errors[error]));
              } else {
                setError(errors);
              }
            }
          } else {
            setError('Something went wrong. Please try again');
          }
        }
        setLoading(false);
      }
    },
    validateOnChange: false,
    validateOnBlur: true,
    validationSchema: resetPasswordSchema,
  });
  return (
    <ModalContainer
      testId="modalForgotPassword"
      title='Enter your email to reset your password'
    >
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          id='email'
          error={formik.errors.email}
          errorTestId='emailError'
          fieldTestId='emailField'
          label='email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.email}
          value={formik.values.email}
        />
        <GradientButton
          testId='submitButton'
          disabled={loading}
          marginBottom={20}
          marginTop={20}
          type='submit'
          title='Reset'
        />
      </form>
      <CancelButtonContainer>
        <CancelButton
          testId='cancelButton'
          onClick={() => {
            if (!loading) {
              setForgotPassword(false);
            }
          }}
        >
          Cancel
        </CancelButton>
      </CancelButtonContainer>
    </ModalContainer>
  );
};

export default ModalForgotPassword;
