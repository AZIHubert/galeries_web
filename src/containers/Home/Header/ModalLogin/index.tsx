import { useFormik } from 'formik';
import * as React from 'react';

import Field from '#components/Field';
import GradientButton from '#components/GradientButton';
import ModalContainer from '#components/ModalContainer';
import RequiredField from '#components/RequiredField';
import SocialMediaButton from '#components/SocialMediaButton';
import TextButton from '#components/TextButton';
import TextSepatator from '#components/TextSeparator';

import { login } from '#helpers/api';
import { loginSchema } from '#helpers/schemas';

import { ForgotPassword } from './styles';

interface ModalLoginI {
  loading: boolean;
  setForgotPassword: React.Dispatch<React.SetStateAction<boolean>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  switchModal: () => void;
}

const initialValues = {
  password: '',
  userNameOrEmail: '',
};

const ModalLogin = ({
  loading,
  setForgotPassword,
  setLoading,
  switchModal,
}: ModalLoginI) => {
  const formik = useFormik({
    initialValues,
    onSubmit: async (values, { setFieldError }) => {
      if (!loading) {
        try {
          const response = await login(values);
          localStorage.setItem('authToken', response.data.token);
          console.log(response.data);
          setLoading(false);
        } catch (err) {
          const { errors } = err.response.data;
          if (typeof errors === 'object') {
            Object.keys(errors).map((error) => setFieldError(error, errors[error]));
          }
          setLoading(false);
        }
      }
    },
    validateOnChange: false,
    validateOnBlur: true,
    validationSchema: loginSchema,
  });

  return (
    <ModalContainer
      testId='loginModal'
    >
      <SocialMediaButton
        action='login'
        disabled={loading}
        marginBottom={12}
        onClick={() => setLoading(true)}
      />
      <SocialMediaButton
        action='login'
        disabled={loading}
        onClick={() => setLoading(true)}
        variant='google'
      />
      <TextSepatator
        marginBottom={10}
        marginTop={10}
        text='or'
      />
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          id='userNameOrEmail'
          error={formik.errors.userNameOrEmail}
          errorTestId='userNameOrEmailError'
          fieldTestId='userNameOrEmailField'
          marginBottom={7}
          label='user name or email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.userNameOrEmail}
          value={formik.values.userNameOrEmail}
        />
        <Field
          disabled={loading}
          id='password'
          error={formik.errors.password}
          errorTestId='passwordError'
          fieldTestId='passwordField'
          label='password'
          marginBottom={15}
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <RequiredField />
        <div>
          <ForgotPassword
            testId='forgotPasswordButton'
            onClick={() => {
              if (!loading) {
                setForgotPassword(true);
              }
            }}
          >
              Forgot your password?
          </ForgotPassword>
        </div>
        <GradientButton
          testId='submitButton'
          disabled={loading}
          marginBottom={15}
          marginTop={15}
          type='submit'
          title='Log in'
        />
      </form>
      <TextButton
        disabled={loading}
        fontSize={0.65}
        justifyContent='center'
        onClick={switchModal}
        testId='switchToSignin'
        text='You donâ€™t have an account yet? click'
        textButton='here'
      />
    </ModalContainer>
  );
};

export default ModalLogin;
