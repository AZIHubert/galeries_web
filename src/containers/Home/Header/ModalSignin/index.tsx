import { useFormik } from 'formik';
import * as React from 'react';
import FacebookLogin, { ReactFacebookLoginInfo, ReactFacebookFailureResponse } from 'react-facebook-login';
import GoogleLogin from 'react-google-login';

import Field from '#components/Field';
import GradientButton from '#components/GradientButton';
import ModalContainer from '#components/ModalContainer';
import RequiredField from '#components/RequiredField';
import SocialMediaButton from '#components/SocialMediaButton';
import TextButton from '#components/TextButton';
import TextSepatator from '#components/TextSeparator';

import {
  loginFacebook,
  loginGoogle,
  signin,
} from '#helpers/api';

import { signinSchema } from '#helpers/schemas';

interface ModalSigninI {
  loading: boolean;
  setAccountCreate: React.Dispatch<React.SetStateAction<boolean>>;
  setError: React.Dispatch<React.SetStateAction<string>>;
  setCurrentEmail: React.Dispatch<React.SetStateAction<string>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  switchModal: () => void;
}

const initialValues = {
  confirmPassword: '',
  email: '',
  password: '',
  userName: '',
};

const ModalSignin = ({
  loading,
  setAccountCreate,
  setCurrentEmail,
  setError,
  setLoading,
  switchModal,
}: ModalSigninI) => {
  const responseFacebook = async (
    faceBookResponse: ReactFacebookLoginInfo | ReactFacebookFailureResponse,
  ) => {
    setLoading(true);
    try {
      const response = await loginFacebook(faceBookResponse);
      localStorage.setItem('authToken', response.data.token);
      localStorage.setItem('expiresIn', response.data.expiresIn);
      setLoading(false);
    } catch (err) {
      if (err.status === 500) {
        setError('something went wrong');
      } else {
        setError(err.data.errors);
      }
    }
    setLoading(false);
  };
  const responseGoogle = async (
    googleResponse: any,
  ) => {
    setLoading(true);
    try {
      const response = await loginGoogle(googleResponse.profileObj);
      localStorage.setItem('authToken', response.data.token);
      localStorage.setItem('expiresIn', response.data.expiresIn);
    } catch (err) {
      if (err.response.status === 500) {
        setError('something went wrong');
      } else {
        setError(err.response.data.errors);
      }
    }
    setLoading(false);
  };

  const formik = useFormik({
    initialValues,
    onSubmit: async (values, { setFieldError }) => {
      if (!loading) {
        setLoading(true);
        try {
          const response = await signin(values);
          console.log(response.data);
          setAccountCreate(true);
          setCurrentEmail(values.email);
          setLoading(false);
        } catch (err) {
          const { errors } = err.response.data;
          if (typeof errors === 'object') {
            Object.keys(errors).map((error) => setFieldError(error, errors[error]));
          }
          setLoading(false);
        }
      }
    },
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: signinSchema,
  });

  return (
    <ModalContainer
      testId='modalSignin'
    >
      <FacebookLogin
        appId="688539228486770"
        fields="email, gender, name, picture.type(large)"
        onClick={() => setLoading(true)}
        callback={responseFacebook}
        render={(renderProps) => (
          <SocialMediaButton
            action='signin'
            disabled={loading}
            marginBottom={12}
            onClick={renderProps.onClick}
          />
        )}
      />
      <GoogleLogin
        clientId="863840240633-tve0cuo6hib6uhgap61j1nkq03k7k5vq.apps.googleusercontent.com"
        buttonText="Login"
        onSuccess={responseGoogle}
        onFailure={(err) => console.log(err)}
        cookiePolicy={'single_host_origin'}
        render={(renderProps) => (
          <SocialMediaButton
            action='signin'
            disabled={loading}
            onClick={renderProps.onClick}
            variant='google'
          />
        )}
      />

      <TextSepatator
        marginBottom={12}
        marginTop={12}
        text='or'
      />
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          id='userName'
          error={formik.errors.userName}
          errorTestId='userNameError'
          fieldTestId='userNameField'
          marginBottom={7}
          label='user name'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.userName}
          value={formik.values.userName}
        />
        <Field
          disabled={loading}
          id='email'
          error={formik.errors.email}
          errorTestId='emailError'
          fieldTestId='emailField'
          marginBottom={7}
          label='email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.email}
          value={formik.values.email}
        />
        <Field
          disabled={loading}
          id='password'
          error={formik.errors.password}
          errorTestId='passwordError'
          fieldTestId='passwordField'
          marginBottom={7}
          label='password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <Field
          disabled={loading}
          id='confirmPassword'
          error={formik.errors.confirmPassword}
          errorTestId='confirmPasswordError'
          fieldTestId='confirmPasswordField'
          marginBottom={15}
          label='confirm password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          touched={formik.touched.confirmPassword}
          type='password'
          value={formik.values.confirmPassword}
        />
        <RequiredField />
        <GradientButton
          testId='submitButton'
          disabled={loading}
          marginBottom={20}
          marginTop={20}
          type='submit'
          title='Sign in'
        />
      </form>
      <TextButton
        disabled={loading}
        fontSize={0.65}
        justifyContent='center'
        onClick={switchModal}
        testId='switchToLogin'
        text='You already have an account? click'
        textButton='here'
      />
    </ModalContainer>
  );
};

export default ModalSignin;
