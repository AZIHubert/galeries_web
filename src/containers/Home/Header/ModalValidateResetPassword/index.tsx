import * as React from 'react';

import ModalContainer from '#components/ModalContainer';
import TextButton from '#components/TextButton';
import ModalTimer from '#components/ModalTimer';

interface ModalValidateResetPasswordI {
  currentEmail: string;
  loading: boolean,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const ModalValidateResetPassword = ({
  currentEmail,
  loading,
  setLoading,
}: ModalValidateResetPasswordI) => {
  const [open, setOpen] = React.useState<boolean>(false);
  const handleClose = () => setOpen(false);
  return (
    <ModalContainer
      testId="modalValidateResetPassword"
      title='Reset your password'
      titleTextAlign='center'
    >
      <p
        data-testid='validateResetPasswordBody'
      >
        To reset your password, click the
        verification button in
        the email we sent to {currentEmail}.
        This helps keep your account secure.
      </p>
      <TextButton
        disabled={loading}
        justifyContent='center'
        onClick={() => {
          setLoading(true);
          setOpen(true);
        }}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
      <ModalTimer
        handleClose={handleClose}
        open={open}
        text='email resend'
      />
    </ModalContainer>
  );
};

export default ModalValidateResetPassword;
