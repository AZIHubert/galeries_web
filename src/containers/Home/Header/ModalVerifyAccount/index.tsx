import * as React from 'react';

import ModalContainer from '#components/ModalContainer';
import TextButton from '#components/TextButton';

interface ModalVerifyAccountI {
  currentEmail: string;
  loading: boolean,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const ModalVerifyAccount = ({
  currentEmail,
  loading,
  setLoading,
}: ModalVerifyAccountI) => (
  <ModalContainer
    testId="modalVerifyAccount"
    title='Verify your email'
    titleTextAlign='center'
  >
    <p
      data-testid='verifyAccountBody'
    >
      To use Galeries, click the verification
      button in the email we sent
      to {currentEmail}. This helps keep
      your account secure.
    </p>
    <TextButton
      disabled={loading}
      justifyContent='center'
      onClick={() => setLoading(true)}
      marginBottom={30}
      marginTop={20}
      text='No email in your inbox or spam folder? Letâ€™s'
      textButton='resend it'
    />
  </ModalContainer>
);

export default ModalVerifyAccount;
