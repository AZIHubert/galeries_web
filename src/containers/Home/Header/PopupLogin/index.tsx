import * as React from 'react';
import { useFormik } from 'formik';

import FacebookButton from '#components/FacebookButton';
import GoogleButton from '#components/GoogleButton';
import { loginSchema } from '#helpers/schemas';

interface PopupLoginI {
  loading: boolean;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
}

const initialValues = {
  userNameOrEmail: '',
  password: '',
};

const PopupLogin = ({ loading, setLoading }: PopupLoginI) => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {
      if (!loading) { setLoading(true); }
    },
    validateOnChange: false,
    validateOnBlur: true,
    validationSchema: loginSchema,
  });

  return (
    <div>
      <FacebookButton
        loading={loading}
        setLoading={setLoading}
        type={'login'}
      />
      <GoogleButton
        loading={loading}
        setLoading={setLoading}
        type={'login'}
      />
      <form onSubmit={formik.handleSubmit}>
        <label htmlFor="userNameOrEmail">
        user name or email
        </label>
        <input
          data-testid='userNameOrEmailField'
          disabled={loading}
          id='userNameOrEmail'
          name='userNameOrEmail'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='text'
          value={formik.values.userNameOrEmail}
        />
        {formik.errors.userNameOrEmail
        && formik.touched.userNameOrEmail
        && (
          <div
            data-testid='userNameOrEmailError'
          >
            {formik.errors.userNameOrEmail}
          </div>)
        }
        <label htmlFor="password">
        password
        </label>
        <input
          data-testid='passwordField'
          disabled={loading}
          id='password'
          name='password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='password'
          value={formik.values.password}
        />
        {formik.errors.password && formik.touched.password && (
          <div
            data-testid='passwordError'
          >
            {formik.errors.password}
          </div>)
        }
        <button
          disabled={loading}
          data-testid='submitButton'
          type='submit'
        >
          {loading ? 'loading' : 'signin'}
        </button>
      </form>
    </div>
  );
};

export default PopupLogin;
