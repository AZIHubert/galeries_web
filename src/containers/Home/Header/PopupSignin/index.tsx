import * as React from 'react';
import { useFormik } from 'formik';

import FacebookButton from '#components/FacebookButton';
import GoogleButton from '#components/GoogleButton';
import { signinSchema } from '#helpers/schemas';

interface PopupSigninI {
  loading: boolean,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const initialValues = {
  confirmPassword: '',
  email: '',
  password: '',
  userName: '',
};

const PopupSignin = ({ loading, setLoading }: PopupSigninI) => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {
      if (!loading) { setLoading(true); }
    },
    validateOnChange: false,
    validateOnBlur: true,
    validationSchema: signinSchema,
  });

  return (
    <div>
      <FacebookButton
        loading={loading}
        setLoading={setLoading}
        type={'signin'}
      />
      <GoogleButton
        loading={loading}
        setLoading={setLoading}
        type={'signin'}
      />
      <div>
        or
      </div>
      <form onSubmit={formik.handleSubmit}>
        <label htmlFor="userName">
        user name *
        </label>
        <input
          data-testid='userNameField'
          disabled={loading}
          id='userName'
          name='userName'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='text'
          value={formik.values.userName}
        />
        {formik.errors.userName && formik.touched.userName && (
          <div
            data-testid='userNameError'
          >
            {formik.errors.userName}
          </div>)
        }
        <label htmlFor="email">
          email *
        </label>
        <input
          data-testid='emailField'
          disabled={loading}
          id='email'
          name='email'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='text'
          value={formik.values.email}
        />
        {formik.errors.email && formik.touched.email && (
          <div
            data-testid='emailError'
          >
            {formik.errors.email}
          </div>)
        }
        <label htmlFor="password">
          password *
        </label>
        <input
          data-testid='passwordField'
          disabled={loading}
          id='password'
          name='password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='password'
          value={formik.values.password}
        />
        {formik.errors.password && formik.touched.password && (
          <div
            data-testid='passwordError'
          >
            {formik.errors.password}
          </div>)
        }
        <label htmlFor="confirmPassword">
          confirm password *
        </label>
        <input
          data-testid='confirmPasswordField'
          disabled={loading}
          id='confirmPassword'
          name='confirmPassword'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='password'
          value={formik.values.confirmPassword}
        />
        {formik.errors.confirmPassword && formik.touched.confirmPassword && (
          <div
            data-testid='confirmPasswordError'
          >
            {formik.errors.confirmPassword}
          </div>)
        }
        <div>
          * Required field
        </div>
        <button
          disabled={loading}
          data-testid='submitButton'
          type='submit'
        >
          {loading ? 'loading' : 'signin'}
        </button>
      </form>
    </div>
  );
};

export default PopupSignin;
