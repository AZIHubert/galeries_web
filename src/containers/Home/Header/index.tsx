import * as React from 'react';

import Modal from '#components/Modal';
import logo from '#ressources/svg/logoG.svg';

import ModalForgotPassword from './ModalForgotPassword';
import ModalLogin from './ModalLogin';
import ModalSignin from './ModalSignin';
import ModalVerifyAccount from './ModalVerifyAccount';
import ModalValidateResetPassword from './ModalValidateResetPassword';

const Header = () => {
  const [accountCreate, setAccountCreate] = React.useState<boolean>(false);
  const [forgotPassword, setForgotPassword] = React.useState<boolean>(false);
  const [loading, setLoading] = React.useState<boolean>(false);
  const [openLogin, setOpenLogin] = React.useState<boolean>(false);
  const [openSignin, setOpenSignin] = React.useState<boolean>(false);
  const [validateResetPassword, setValidateResetPassword] = React.useState<boolean>(false);
  const [currentEmail, setCurrentEmail] = React.useState<string>('');

  const handleClickLogin = () => {
    if (!loading) {
      setOpenSignin(false);
      setOpenLogin(!openLogin);
    }
  };
  const handleClickSignin = () => {
    if (!loading) {
      setOpenLogin(false);
      setOpenSignin(!openSignin);
    }
  };
  const handleCloseModal = () => {
    if (!loading) {
      setAccountCreate(false);
      setForgotPassword(false);
      setOpenLogin(false);
      setOpenSignin(false);
      setValidateResetPassword(false);
      setCurrentEmail('');
    }
  };
  const switchToValidateResetPassword = () => {
    if (!loading) {
      setForgotPassword(false);
      setValidateResetPassword(true);
    }
  };

  const LoggerModal = accountCreate ? (
    <ModalVerifyAccount
      currentEmail={currentEmail}
      loading={loading}
      setLoading={setLoading}
    />
  ) : (
    <ModalSignin
      loading={loading}
      setAccountCreate={setAccountCreate}
      setCurrentEmail={setCurrentEmail}
      setLoading={setLoading}
      switchModal={handleClickLogin}
    />
  );
  const SignerModal = () => {
    if (forgotPassword) {
      return (
        <ModalForgotPassword
          setCurrentEmail={setCurrentEmail}
          setForgotPassword={setForgotPassword}
          setLoading={setLoading}
          loading={loading}
          switchModal={switchToValidateResetPassword}
        />
      );
    }
    if (validateResetPassword) {
      return (
        <ModalValidateResetPassword
          currentEmail={currentEmail}
          loading={loading}
          setLoading={setLoading}
        />
      );
    }
    return (
      <ModalLogin
        loading={loading}
        setLoading={setLoading}
        setForgotPassword={setForgotPassword}
        switchModal={handleClickSignin}
      />
    );
  };

  return (
    <header>
      <img
        src={logo}
        alt="header logo"
      />
      <button
        onClick={handleClickSignin}
        data-testid='openSignin'
      >
        signin
      </button>
      <button
        data-testid='openLogin'
        onClick={handleClickLogin}
      >
        login
      </button>
      <Modal
        open={openLogin || openSignin}
        handleClose={handleCloseModal}
      >
        {openLogin && SignerModal()}
        {openSignin && LoggerModal}
      </Modal>
    </header>
  );
};

export default Header;
