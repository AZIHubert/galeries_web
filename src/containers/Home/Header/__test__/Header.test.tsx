import {
  fireEvent,
  render,
  screen,
  act,
} from '@testing-library/react';
import * as React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Header from '../index';

describe('Header', () => {
  beforeAll(() => {
    // @ts-ignore
    ReactDOM.createPortal = jest.fn((element) => element);
  });
  afterEach(() => {
    // @ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  it('renders without crashing', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render verify account modal if account is created', async () => {
    act(() => {
      render(<Header />);
    });
    const openSignin = screen.getByTestId('openSignin');
    fireEvent.click(openSignin);
    const confirmPasswordField = screen.getByTestId('confirmPasswordField');
    const emailField = screen.getByTestId('emailField');
    const passwordField = screen.getByTestId('passwordField');
    const submitButton = screen.getByTestId('submitButton');
    const userNameField = screen.getByTestId('userNameField');
    fireEvent.change(confirmPasswordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(emailField, { target: { value: 'user@email.com' } });
    fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(userNameField, { target: { value: 'user' } });
    fireEvent.click(submitButton);
    await screen.findByTestId('modalVerifyAccount');
    const modalSignin = screen.queryByTestId('modalSignin');
    expect(modalSignin).toBeNull();
  });
  it('should switch between login and signin modal', () => {
    const { getByTestId } = render(<Header />);
    const openLogin = getByTestId('openLogin');
    fireEvent.click(openLogin);
    const switchToSignin = screen.getByTestId('switchToSignin');
    fireEvent.click(switchToSignin);
    const modalSignin = screen.queryByTestId('modalSignin');
    expect(modalSignin).not.toBeNull();
    const switchToLogin = screen.getByTestId('switchToLogin');
    fireEvent.click(switchToLogin);
    const loginModal = screen.queryByTestId('loginModal');
    expect(loginModal).not.toBeNull();
  });
});
