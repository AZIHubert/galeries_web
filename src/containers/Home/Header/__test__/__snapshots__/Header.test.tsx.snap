// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="sc-dIUggk bzTsUN"
>
  <div
    className="sc-hHftDr bGHNmb"
  >
    <img
      alt="header logo"
      className="sc-dmlrTW bPBhZW"
      src={Object {}}
    />
    <div
      className="sc-idOhPF cbJGzu"
    >
      <button
        className="sc-bdfBwQ kjBTJr"
        data-testid="openSignin"
        onClick={[Function]}
      >
        Sign in
      </button>
      <button
        className="sc-bdfBwQ ctZNxY"
        data-testid="openLogin"
        onClick={[Function]}
      >
        Log in
      </button>
    </div>
  </div>
</header>
`;
